packages/core/rfcs/use-user-address-factory.md

```diff
@@ -0,0 +1,128 @@
+# UseUserAddressFactory with Commerce Tools composable use case
+
+## Motivation
+
+We'd like to separate a logic for the user addresses according to single responsibility principle. It's also more agnostic approach.
+
+## Factory interface
+
+```TS
+// factory schema
+export interface UseUserAddressFactory<ADDRESS, SEARCH_PARAMS> {
+  addresses: ComputedProperty<ADDRESS[]>;
+  totalAddresses: ComputedProperty<number>;
+  searchAddresses: (params?: SEARCH_PARAMS) => Promise<void>;
+  addShippingAddress: (address: ADDRESS) => Promise<void>;
+  addBillingAddress: (address: ADDRESS) => Promise<void>;
+  getShippingAddresses: () => ADDRESS[];
```

Comment:
that shouldnt be a getX function (or if it would remain like that it should be a getter but probably that wasn;t your intention) since it collides with getters naming.

`billingAddresses` seems like a better way of naming it BUT then you have a duplication of data (`addressess` will contain a superset of `billingAddresses`). Can we have more than 1 assigned billing addr/shipping adds since you used a plural form here?

If we don't shouldn't that be just a computed property? is it fetched separately?

Another option would be to just have a getter `getBillingAddresses(address: ADDRESS[])`

Reply:
Computed value makes sens for me